cli/cli/out/cli_writers.go

```diff
@@ -40,13 +40,25 @@ func GetErr() io.Writer {
 }
 
 func PrintOutLn(msg string) {
+	printLogsToFile(msg)
 	if _, printErr := fmt.Fprintln(std.out, msg); printErr != nil {
 		logrus.Errorf("Error printing message to StdOut. Message was:\n%s\nError was:\n%v", msg, printErr.Error())
 	}
 }
 
 func PrintErrLn(msg string) {
+	printLogsToFile(msg)
 	if _, printErr := fmt.Fprintln(std.err, msg); printErr != nil {
 		logrus.Errorf("Error printing message to StdErr. Message was:\n%s\nError was:\n%v", msg, printErr.Error())
 	}
 }
+
+func printLogsToFile(msg string) {
+	fileLogger, err := GetFileLogger()
+	if err != nil {
+		logrus.StandardLogger().Warnf("Error using the file logger as it failed with following error: %v. "+
```

Comment:
Yes we need this logger; because i have added hook to logrus ( which trigger filelogger in the background to add logs to the file). It will potentially create a recursive loop. added a comment here.

Reply:
Not sure I fully understand what this hook thingy is, but does it mean that everywhere we do logrus.Infof in the CLI, we have the add this `.StandardLogger()`?

Reply:
we have to use standardLogger only when we are logging fileLogger failure to be more precise. The main rationale was to have no change in current development. I do have a FLUP to hide this failure behind High Order Function so that developers don't need to think about it. We can discuss this over a call, if need mores clarification @gbouv